www/js/diary/services.js

```diff
@@ -133,6 +131,45 @@ angular.module('emission.main.diary.services', ['emission.services'])
       return '';
     }
   }
+  dh.getArrowClass = function(i) {
+    if (i == -1) {
+      return 'icon ion-arrow-down-c';
+    } else if (i == 0) {
+      return '';
+    } else {
+      return 'icon ion-arrow-up-c';
+    }
+
+  }
+  dh.getLongerOrShorter = function(trip, id) {
+    var ctrip = CommonGraph.findCommon(id);
+    if (!angular.isUndefined(ctrip)) {
+      var cDuration = ctrip.durations[0];
```

Comment:
This I will fix. I think that you want to take the average duration and not the first duration.